src/util/dataIntegrityCheck.js

```diff
@@ -53,7 +55,7 @@ export const dataIntegrityCheck = path => {
 
   // recreate thoughts missing in thoughtIndex
   for (const child of (contextIndex[encoded] || [])) { // eslint-disable-line fp/no-loops,fp/no-let
-    const childExists = exists(child.value, thoughtIndex)
+    const childExists = exists(thoughtIndex, child.value)
```

Comment:
Should selectors take the full state, or a slice?

@anmolarora1 We had talked about this and I think decided on full state, but do you still think that's the right thing? It seems odd to pass the full state when it only needs a slice.

Reply:
@umair-khanzada I discussed with @anmolarora1, and we would like to pass the whole `state` object to each selector (or least in the same shape as the full state, e.g. `{ thoughtIndex }` rather than `thoughtIndex`).

Reply:
Got it